### PR TITLE
Issue/851 – Add option to blacklist URLs

### PR DESCRIPTION
**Pull request by DrewAPicture**
Issue: #851

Updated and moved from #874.

### CODE REVIEW

File: includes/referral-functions.php [hunk 1]
```diff
@@ -241,3 +241,46 @@ function affwp_count_referrals( $affiliate_id = 0, $status = array(), $date = ar
 	return affiliate_wp()->referrals->count( $args );
 
 }
+
+/**
+ * Retrieve an array of banned URLs
+ *
+ * @since 1.7.5
```
Comment (michaelbeil) [inline includes/referral-functions.php]: version 1.9. same below.

File: includes/class-tracking.php [hunk 2]
```diff
@@ -209,27 +209,34 @@ public function track_visit() {
...
+					$this->log( sprintf( 'Visit #%d recorded for affiliate #%d in track_visit()', $visit_id, $affiliate_id ) );
+				}
 
-			echo $visit_id; exit;
+				echo $visit_id;
 
-		} else if( ! $is_valid ) {
+				exit;
+			} else {
+				die( '-2' );
```
Comment (pippinsplugins) [inline includes/class-tracking.php]: Let's add a debug log for when banned URLs are hit

Comment (DrewAPicture) [inline includes/class-tracking.php]: This is done. Also updated the source branch.

File: includes/class-tracking.php [hunk 3]
```diff
@@ -371,19 +384,21 @@ public function fallback_track_visit() {
 		$visit_id     = $this->get_visit_id();
 
 		if( $is_valid && ! $visit_id ) {
+			if( ( ! empty( $_SERVER['HTTP_REFERER'] ) && ! affwp_is_url_banned( sanitize_text_field( $_POST['referrer'] ) ) ) || empty( $_SERVER['HTTP_REFERER'] ) ) {
```
Comment (pippinsplugins) [inline includes/class-tracking.php]: `$_POST['referrer']` should be `$_SERVER['HTTP_REFERER']`

**Pull request merged**